src/containers/random-pick/random-pick-playground-provider.tsx/random-pick-playground-provider.tsx

```diff
@@ -0,0 +1,188 @@
+import {
+  createContext,
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { useRandomPickState } from '../random-pick-provider/random-pick-provider.hooks';
+import { MODAL_STATE_TYPES } from './random-pick-playground-provider.constans';
+
+type ModalStateType =
+  (typeof MODAL_STATE_TYPES)[keyof typeof MODAL_STATE_TYPES];
+
+export type WinnersType = {
+  pickListId: string;
+  pickListValue: string;
+  isflipped: boolean;
+};
+
+type RandomPickPlaygroundState = {
+  numberOfPick: number;
+  winners: WinnersType[];
+  temporaryPickList: string[];
+  modalState: ModalStateType;
+  maxNumberOfPick: number;
+  forceRender: number;
+};
+export const RandomPickPlaygroundStateContext =
+  createContext<RandomPickPlaygroundState | null>(null);
+
+type RandomPickPlaygroundAction = {
+  selectModalState: Dispatch<SetStateAction<ModalStateType>>;
+  runPick: (newNumberOfPick?: number) => void;
+  resetPick: () => void;
+  handleCardFlip: (id: string) => void;
+};
+
+export const RandomPickPlaygroundActionContext =
+  createContext<RandomPickPlaygroundAction | null>(null);
+
+export default function RandomPickPlaygroundProvider({
+  children,
+}: PropsWithChildren<{}>) {
+  const [numberOfPick, setNumberOfPick] = useState(1);
+  const [winners, setWinners] = useState<WinnersType[]>([]);
+  const [forceRender, setForceRender] = useState(1);
+  const [temporaryPickList, setTemporaryPickList] = useState([]);
+
+  const [modalState, setModalState] = useState<ModalStateType>(
+    MODAL_STATE_TYPES.noModal,
+  );
+  const numberOfWinner = useRef(0);
+  const cardMixRef = useRef<NodeJS.Timeout | null>(null);
+
+  const {
+    pickType,
+    pickList,
+    options: { isExcludingSelected },
+  } = useRandomPickState();
+
+  useEffect(() => {
+    setTemporaryPickList(pickList[pickType].map((e) => e.value));
+  }, [pickList, pickType]);
+
+  useEffect(() => {
+    if (modalState === MODAL_STATE_TYPES.setPickNumberModal) {
+      cardMixRef.current = setInterval(() => {
+        setTemporaryPickList((prev) =>
+          [...prev].sort(() => Math.random() - 0.5),
+        );
+      }, 100);
+    } else {
+      clearInterval(cardMixRef.current);
+    }
+  }, [modalState]);
```

Comment:
```suggestion
  useEffect(() => {
    if (modalState === MODAL_STATE_TYPES.setPickNumberModal) {
      cardMixRef.current = setInterval(() => {
        setTemporaryPickList((prev) =>
          [...prev].sort(() => Math.random() - 0.5),
        );
      }, 100);
    } else {
      clearInterval(cardMixRef.current);
    }
    
    return () => clearInterval(cardMixRef.current);
  }, [modalState]);
```

`useEffect` 클린업이 필요할 것 같네요!